Remove extra ';' in src/xlsparse.c which turned for loop in xlsparse into a
buffer overflow (Closes: #692076), applies patch by Olly Betts
<olly@survex.com>.

diff --git a/src/xlsparse.c b/src/xlsparse.c
index 886ffff..46da488 100644
--- a/src/xlsparse.c
+++ b/src/xlsparse.c
@@ -589,8 +589,8 @@ static char FormatIdxUsed[NUMOFDATEFORMATS];
 
 void CleanUpFormatIdxUsed() {
 	int i;
-	for (i=0;i<NUMOFDATEFORMATS; i++);
-	FormatIdxUsed[i]=0;
+	for (i=0;i<NUMOFDATEFORMATS; i++)
+		FormatIdxUsed[i]=0;
 }
 
 /* 
